src/module_2/Module_2_EDA_Ex1.md

```diff
@@ -0,0 +1,3152 @@
+grouped_data = data.groupby([data['order_date'].dt.year, data['order_date'].dt.month])
+monthly_data = grouped_data.agg({
+    'product_order_count' : 'sum',
+    'product_abandoned_count' : sum
+    })
+
+for year in monthly_sales.index.get_level_values(0).unique():
+    monthly_data.loc[year].plot(kind='bar', stacked = True)
+    plt.title(f'Number of Products Sold and Abandoned Each Month in the year {year}')
+    plt.xlabel('Order Date')
+    plt.ylabel('Products Sold')
+    plt.xticks(ticks=range(len(monthly_sales.loc[year])), labels=[f'{year}-{month:02d}' for month in monthly_sales.loc[year].index], rotation=45)
+    plt.legend(['Ordered products', 'Abandoned products'])
+    plt.show()
+```
+
+    /tmp/ipykernel_154450/1745198578.py:3: FutureWarning: The provided callable <built-in function sum> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
+      monthly_data = grouped_data.agg({
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_104_1.png)
+    
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_104_2.png)
+    
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_104_3.png)
+    
+
+
+We can observe that, comparing to the number of products that are sold, not that many products are abandoned. Furthermore, we can see that in 2020, there are not many products registered as abandoned. On the other hand, in the last months of 2021, many products were, specially in December.
+
+**c. How many of the products sold are regular ones**
+
+
+```python
+# Function to calculate the count of regular items ordered
+def count_regulars(row):
+    # Set intersection to find regular items in the order
+    return len(set(row['ordered_items']) & set(row['regular_items']))
+```
+
+
+```python
+data['regular_items_count'] = data.apply(count_regulars, axis=1)
+
+
+monthly_regulars = data.groupby([data['order_date'].dt.year, data['order_date'].dt.month])['regular_items_count'].sum()
+
+monthly_regulars.plot(kind='bar')
+plt.title('Number of Regular Products Sold Each Month')
+plt.xlabel('Month')
+plt.ylabel('Count of Regular Products Sold')
+plt.xticks(ticks=range(len(monthly_regulars)), labels=[f'{year}-{month:02d}' for year, month in monthly_regulars.index], rotation=90)
+plt.show()
+```
+
+    /tmp/ipykernel_154450/3098253969.py:1: SettingWithCopyWarning: 
+    A value is trying to be set on a copy of a slice from a DataFrame.
+    Try using .loc[row_indexer,col_indexer] = value instead
+    
+    See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+      data['regular_items_count'] = data.apply(count_regulars, axis=1)
+
+
+
+    
+![png](Module_2_EDA_Ex1_files/Module_2_EDA_Ex1_108_1.png)
+    
+
+
+We can observe that the distribution is very similar to the one of the sold products.
```

Comment:
Then it woukld be intersetinsg to know the %, to see if the increase comes from an increase in sales or in regularity of proeducts